crates/vm/src/arch/execution_mode/metered/memory_ctx.rs

```diff
@@ -261,15 +335,16 @@ impl<const PAGE_BITS: usize> MemoryCtx<PAGE_BITS> {
             let nodes = (((1 << PAGE_BITS) - 1) + (merkle_height - PAGE_BITS)) as u32;
             // SAFETY: merkle_tree_idx is guaranteed to be in bounds
             unsafe {
-                *trace_heights.get_unchecked_mut(poseidon2_idx) += nodes * 2;
-                *trace_heights.get_unchecked_mut(merkle_tree_idx) += nodes * 2;
+                *trace_heights.get_unchecked_mut(poseidon2_idx) +=
+                    nodes * page_access_count as u32 * 2;
+                *trace_heights.get_unchecked_mut(merkle_tree_idx) +=
+                    nodes * page_access_count as u32 * 2;
```

Comment:
i think this was a bug earlier where the update should've been `nodes * page_access_count * 2` instead of `nodes * 2`